### PR TITLE
crkbd/keymaps/rs: add rs keymap to corne keyboard and adapt others

### PR DESCRIPTION
**Pull request by rs**
## Description

Adds a new code friendly keymap for corne keyboard + update all my keymaps to work with 40% layouts.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

### CODE REVIEW

File: keyboards/crkbd/keymaps/rs/keymap.c [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+#define KC_RPGT RPGT // ) or > with shift
+#define KC_NEQL NEQL // !=
+#define KC_CODE MO(_CODE)
+#define KC_FN   MO(_FN)
+#define KC_RST  RESET
+#define KC_CTRA LCTL(KC_A)
+#define KC_CTRE LCTL(KC_E)
+// Brightness
+#define KC_BRUP KC_PAUS
+#define KC_BRDN KC_SLCK
```
Comment (fauxpark) [inline keyboards/crkbd/keymaps/rs/keymap.c]: These brightness aliases for `KC_PAUS` and `KC_SLCK` are defined in QMK already as `KC_BRMU/BRMD`. But even better, there are media keys that, to my knowledge, work on Windows, macOS *and* Linux: [`KC_BRIU/BRID`](https://docs.qmk.fm/#/keycodes_basic?id=media-keys).

Comment (rs) [reply keyboards/crkbd/keymaps/rs/keymap.c]: Thanks, that's good to know.

File: users/rs/rs.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+__attribute__ ((weak))
+void rgb_mod_changed_keymap(void) {
+}
+
+__attribute__ ((weak))
+void keylog_set_keymap(uint16_t keycode, keyrecord_t *record) {
+}
+
+void press_key(uint16_t key) {
```
Comment (drashna) [inline users/rs/rs.c]: We actually have a built in `tap_code` function that does this.

File: users/rs/rs.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      } else {
+        // )
+        press_two_keys(KC_LSFT, KC_0);
+      }
+    }
+    return false;
+#ifdef RGBLIGHT_ENABLE
+  case RGB_MOD:
+      if (record->event.pressed) {
+        rgblight_step();
```
Comment (drashna) [inline users/rs/rs.c]: Actually, if you return true, and check on `!record->event.pressed`, you shouldn't need this, I think

File: users/rs/rs.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
+#pragma once
+#include "quantum.h"
+#include "action.h"
```
Comment (drashna) [inline users/rs/rs.h]: I don't think that you actually need this here. 

```suggestion
```

File: users/rs/rs.c (hunk 3 above)
Comment (rs) [reply users/rs/rs.c]: The problem is that `rgb_mod_changed_keymap` has to be call after the fact.

Comment (drashna) [reply users/rs/rs.c]: Well, you could use `!record->event.pressed` instead, so it happens on upstroke, but I can understand that

**Comment by rs** [comment]
Do you need anything from me in order to merge this?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**